### PR TITLE
Structure view enhancement

### PR DESCRIPTION
**Pull request by jooohhn**
## Summary
- [x] Removed hardcoded structure view
- [x] Can dynamically view table structure
- [ ] Can edit table structure

## Pictures
- image

- image

### CODE REVIEW

File: app/components/Structure.js [hunk 1]
```diff
@@ -8,52 +8,44 @@ import 'brace/snippets/sql';
...
-    this.state = {
-      tableName: 'FooTable'
-    };
+    // Needs state.loading or else props.selectedName and state.tableKeys
+    // won't align
+    this.state = { tableKeys: null, loading: true };
+  }
+
+  async componentDidMount() {
+    this.setState({ tableKeys: await this.props.tableKeysPromise });
```
Comment (amilajack) [inline app/components/Structure.js]: No need to call setState multiple times. Call as a single object. ex:

```js
// Before
setState({ foo })
setState({ bar })

// After
setState({ foo, bar })
```

Do this for all other `setState` calls

File: app/components/Structure.js [hunk 2]
```diff
@@ -64,7 +56,7 @@ export default class Structure extends Component {
           theme="xcode"
           name="querybox"
           ref="queryBoxTextarea"
-          value={"SELECT * FROM 'users'"}
+          value={'PLACEHOLDER TEXT'}
```
Comment (amilajack) [inline app/components/Structure.js]: Do we need the `AceEditor` component here?

**Comment by amilajack** [comment]
Damnnnn this is sweet! 🎉 🎉 🎉  Good work on this man 👊

**Comment by amilajack** [comment]
Seems like the table values are being casted from (boolean or string to a number).
screen shot 2017-08-24 at 11 04 07 am

**Comment by amilajack** [comment]
What do you think about changing the button to use the antd styles?

screen shot 2017-09-19 at 6 46 33 pm

**Comment by amilajack** [comment]
screen shot 2017-09-19 at 6 48 55 pm

Any ideas why sqlite is returning a 2 for the primary key here?

**Comment by jooohhn** [comment]
Yeah I design for now, I just wanted something functional and that we can sub out in the redesign

**Comment by jooohhn** [comment]
The PR is still being worked on. I'm pretty sure I'll have it worked out before the end of this week though

**Pull request merged**